### PR TITLE
Mini-station vault

### PR DESCRIPTION
**Pull request by MadmanMartian**
An idea shot around in the discord - A mini-station you can potentially spawn on.

image

Currently has spots for: 1 medical doctor, 1 engineer, and 1 botanist.

Will likely have to have a preference in place for if a player wants to spawn off-station.

### CODE REVIEW

**Comment by DrSnips** [comment]
Why bother having the radio transmitter when there's no wall radios?
Also take a look at how big the power drain is on that APC. I assume it's running the whole station so you might want to bump up the power cell inside it.

**Comment by vampirebat74** [comment]
APC for every department, newscasters, wall speakers. Add 'em

**Comment by spoonylove** [comment]
This reminds of the vault on the fallout server. Maybe make them a small syndie safe house. Non combat syndie crew. Do like deff make all the equipment black.

**Comment by Kurfursten** [comment]
Snips is right, you might be slightly confused. The radio transmitter is for setting up media to play over speakers, you only need a relay if you just want headsets to work. It's redundant if the whole station is on a single area/APC since you can just use the juke normally.

**Comment by Durgen** [comment]
more windows?
maybe some external scaffolding?

**Comment by SonixApache** [comment]
Add one (1) oxygen canister. Repeated use of the external airlock would result in loss of pressure over time.

**Comment by MadmanMartian** [comment]
The transmitter's for the jukebox

**Comment by SonixApache** [comment]
Transmitter's kind of pointless then, apart from cooking you whenever you're in the engineering section.
Unless you want your little station to double as a DJ station knockoff.

**Comment by MadmanMartian** [comment]
Update: Atmospherics added. Image updated.

There's some sprite abnormalities on the vents and scrubbers, that's because I added some stuff to the sprites local-side so I could tell their directions in the map editor.

**Comment by Arthurdentist** [comment]
I can't remember if Celt ever got around to coding this, but back when snow map was an active project I suggested to him that he should code a way to exclude specific vaults from spawning on specific maps. This would be useful here, as it would probably be a good idea to exclude this vault from spawning during a Packed Station round, where the population is already very low and having people spawn in spess and not on the station would cause real issues for the round.

**Comment by MadmanMartian** [comment]
good call

**Comment by Kevak** [comment]
It might be wise to prevent roundstart antag for those who spawn on the ministation in order to prevent situations such as 'vampire stranded in space' or 'cult's down a member' or 'whoops, revhead spawned offstation'

Also botany needs a biogenerator and seed extractor and you might wanna consider a microwave and other kitchen tools such as a food processor and a condimaster.

Consider solars. PACMAN generators are limited in power production due to lack of plasma.

Rwalls and glassless door around the transmitter room to avoid irradiating engineering. 

Do the interior walls of the vault really need to be rwalls? Would normal walls not suffice?

APC for every room/section. RCD might be wise. Maybe a nearby asteroid/pickaxe so engineering can do 'something' and have a way to expand the vault?

Does the MD have a biosuit?

Newscaster.

Consider a small shuttlepod/destination disc.

Advanced medical scanner. Maybe replace a sleeper with it.

Consider gas miners of the oxygen and nitrogen varieties so the engineer can tinker with atmos and have gas to replace lost air in the event of breaches.

Consider a bar/bar machines.

Consider exterior scaffolding for navigation and aesthetic.

Firelocks.

Consider more exterior windows. Consider said windows with firelocks.

Consider a circuit imprinter circuit and old-R&D console circuit.

**Comment by Kurfursten** [comment]
I don't think those (cash based) vending machines will work due to lack of accounts database on the z level

**Comment by ghost** [comment]
I would personally suggest making this a ghost role, locking those who spawn on it to the z-level it spawns on, and making it a little less... complete. 
We really don't want more people spawning off station at round start. Especially when we don't have constantly medium-high pop rounds. 

If you have never played on /tg/, I suggest doing it just once to see their version of something like this. It's very fun to try it, especially if you're stuck dead because your body is destroyed or you're somehow else irretrievable. 

I say this a lot but there's one thing they do right over there and it's the ghost roles and the potential for bringing players who are locked out of a round back in to play and have some fun. For it to work on /vg/ though, I think it would need changing slightly because of the hellscape it is over there with pseudo-antagonist rules and whatever, but it's fun, it's better than being a derelict MoMMI and gives people who lose out on a round something else to do instead of just closing and finding another game to play.

**Comment by Kurfursten** [comment]
Reinforced walls are bad because they hurt customization of the interior

Don't forget botany disks

**Comment by MadmanMartian** [comment]
currently trying to get into pipe-mapping

Expected: 
image

Actual: 
widdershins town 11 2018-12-10 143112

I.Dont.Underastandu

**Comment by Tehsapper** [comment]
remove all but two unit storages so this could be a fun mini-gamemode.
otherwise this honestly serves no purpose except being a 3-man circlejerk box on already low-pop server. can't expand without mining, gas miners or r&d, so you're pretty much forced to leave it and join the rest of players.

**Comment by MadmanMartian** [comment]
So, some people want MORE stuff, and others want less?

File: maps/randomvaults/mini_station.dmm [hunk 1]
```diff
@@ -60,121 +60,133 @@
...
+"cO" = (/obj/structure/table{pixel_x = 0; pixel_y = 0},/obj/item/weapon/switchtool/surgery,/obj/item/weapon/storage/firstaid/toxin,/obj/item/stack/sheet/charcoal{pixel_w = 5},/turf/simulated/floor{icon_state = "blue"; dir = 6},/area/vault/mini_station)
 "cP" = (/obj/machinery/light{dir = 8; icon_state = "ltube1"},/turf/simulated/floor,/area/vault/mini_station)
 "cQ" = (/obj/machinery/light{dir = 4},/turf/simulated/floor,/area/vault/mini_station)
 "cR" = (/obj/machinery/light{dir = 8},/obj/item/weapon/bikehorn,/turf/simulated/floor/engine,/area/vault/mini_station)
 "cS" = (/obj/structure/table/glass,/obj/item/weapon/folder/yellow,/turf/simulated/floor,/area/vault/mini_station)
 "cT" = (/obj/structure/table/glass,/obj/machinery/light{dir = 4},/obj/item/weapon/crowbar,/turf/simulated/floor,/area/vault/mini_station)
+"cU" = (/obj/machinery/newscaster{dir = 4; pixel_x = -28},/turf/simulated/floor,/area/vault/mini_station)
+"cV" = (/obj/structure/sign/greencross,/turf/simulated/wall,/area/vault/mini_station)
+"cW" = (/obj/machinery/chem_master{pixel_z = 1; pixel_w = 5},/turf/simulated/floor{dir = 5; icon_state = "blue"},/area/vault/mini_station)
```
Comment (DamianX) [inline maps/randomvaults/mini_station.dmm:153]: \>pixel_z
\>pixel_w
\>pixel_anything on a chem master
???

**Comment by Lunstra** [comment]
I quite like the idea of this as a ghost role tbh. Partially for the aspect of getting perma dead people who don't want to be greentide/brainslug/a souless automaton slave, back in a round in some capacity, and partially for observers who want to practice something you can't as a MoMMI, like surgery, with a more learned player teaching them in peace. It'd allow for some more organic migrant worker bus stuff for getting said players back on the main station too. It probably should have solars though since a pacman is a limited supply of power, although you could presumably just make a crank charger and charge the APC with that I guess, and maybe a limited supply of building materials for some minor expansion.

**Pull request closed without merging**